posts/2022/02/making-gpus-accessible-to-pydata-ecosystem-via-array-api.md

```diff
@@ -0,0 +1,507 @@
+<!--
+.. title: Making GPUs accessible to the PyData Ecosystem via the Array API Standard.
+.. slug: making-gpus-accessible-to-pydata-ecosystem-via-array-api
+.. date: 2022-02-01 21:07:02 UTC-06:00
+.. author: Amit Kumar
+.. tags: GPU, SciPy, NumPy, CuPy, scikit-learn, scikit-image, Array API
+.. category:
+.. link:
+.. description:
+.. type: text
+.. previewimage: /images/2022/02/array_api_workflow.svg
+-->
+
+
+
+GPUs have become an essential part of the scientific computing stack and with
+the advancement in the technology around GPUs and the ease of accessing a GPU
+in the cloud or on-prem, it is in the best interest of the PyData community to spend
+time and effort to make GPUs accessible for the users of PyData libraries. A
+typical user in the PyData ecosystem is quite familiar with the APIs of libraries
+like SciPy, scikit-learn, and scikit-image -- and at the moment these
+libraries are largely limited to single-threaded operations on CPU. In this
+blog post I will talk about how we can use the [Python Array API Standard](https://data-apis.org/array-api/latest/)
+with the fundamental libraries in the PyData ecosystem along with CuPy for
+making GPUs accessible to the users of these libraries. With the introduction
+of that standard by the [Consortium for Python Data API Standards](https://data-apis.org/)
+and its adoption mechanism in [NEP 47](https://numpy.org/neps/nep-0047-array-api-standard.html)
+it is now possible to write code that is portable between NumPy and other array/tensor
+libraries that adopt the Array API Standard. We will also discuss the workflow and
+challenges for actually achieving such portability.
+
+<!-- TEASER_END -->
+
+## Motivation and Goal
+
+The motivation for this exercise comes from the fact that having GPU-specific code
+in SciPy and scikits will be too-specialized and very hard to maintain. The goal of
+this demo is to (a) to show things working in a real-world example on GPUs, and
+(b) demonstrate that there is a significant performance benefit. We'll be using
+CuPy as the GPU library for this demo. Since CuPy has support for NVIDIA GPUs
+as well as AMD’s ROCm GPU platform, this demo will work on either of those GPUs.
+
+The ultimate goal is to have SciPy, scikit-learn and scikit-image accept any type
+of array, so they can work with CuPy, PyTorch, JAX, Dask and other libraries.
+This project is part of a larger [vision sponsored by
```

Comment:
I tried to use "demo" and "work" for the Greek coins demo in this blog post, and "project" for the overall project sponsored by AMD in a consistent way. So I'd like to give this sentence a tweak like that. Other than that it reads well.

Reply:
Ah yes, makes sense